File: uri/ad-hock/kings_poker/1337.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    int a, b, c;
    
    while (cin >> a >> b >> c && (a || b || c)) {
        int cards[3] = {a, b, c};
        sort(cards, cards + 3);
        
        if (cards[0] == cards[1] && cards[1] == cards[2]) {
            int rank = cards[0];
            
            if (rank == 13) {
                cout << "*" << endl;
            } else {
                cout << (rank + 1) << " " << (rank + 1) << " " << (rank + 1) << endl;
            }
        } else if (cards[0] == cards[1] || cards[1] == cards[2]) {
            int pairRank, single;
            if (cards[0] == cards[1]) {
                pairRank = cards[0];
                single = cards[2];
            } else {
                pairRank = cards[1];
                single = cards[0];
            }
            
            int nextCard = single + 1;
            if (nextCard == pairRank) {
                nextCard++;
            }
            
            if (nextCard <= 13) {
                cout << min(nextCard, pairRank) << " " << pairRank << " " << max(nextCard, pairRank) << endl;
            } else if (pairRank < 13) {
                cout << "1 " << (pairRank + 1) << " " << (pairRank + 1) << endl;
            } else {
                cout << "1 1 1" << endl;
            }
        } else {
            cout << "1 1 2" << endl;
        }
    }
    
    return 0;
}

File: uri/ad-hock/kings_poker/1337.rs
use std::io::{self, BufRead};
use std::cmp::{min, max};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    
    while let Some(Ok(line)) = lines.next() {
        let nums: Vec<i32> = line
            .split_whitespace()
            .map(|s| s.parse().unwrap())
            .collect();
        
        let (a, b, c) = (nums[0], nums[1], nums[2]);
        
        if a == 0 && b == 0 && c == 0 {
            break;
        }
        
        let mut cards = vec![a, b, c];
        cards.sort();
        
        if cards[0] == cards[1] && cards[1] == cards[2] {
            let rank = cards[0];
            if rank == 13 {
                println!("*");
            } else {
                println!("{} {} {}", rank + 1, rank + 1, rank + 1);
            }
        } else if cards[0] == cards[1] || cards[1] == cards[2] {
            let (pair_rank, single) = if cards[0] == cards[1] {
                (cards[0], cards[2])
            } else {
                (cards[1], cards[0])
            };
            
            let mut next_card = single + 1;
            if next_card == pair_rank {
                next_card += 1;
            }
            
            if next_card <= 13 {
                println!("{} {} {}", min(next_card, pair_rank), pair_rank, max(next_card, pair_rank));
            } else if pair_rank < 13 {
                println!("1 {} {}", pair_rank + 1, pair_rank + 1);
            } else {
                println!("1 1 1");
            }
        } else {
            println!("1 1 2");
        }
    }
}
